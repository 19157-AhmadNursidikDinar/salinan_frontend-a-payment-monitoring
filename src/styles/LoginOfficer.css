@import url("https://fonts.googleapis.com/css2?family=Poppins:wght@200&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Poppins:wght@500&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Roboto:wght@100;300;400;700&display=swap");

.container {
  display: flex;
  flex-direction: row;
  width: 100%;
  background-color: grey;
}
.left {
  flex: 1;
  display: flex;
  width: 50%;
  justify-content: center;
  align-items: center;
  background-color: white;
}
.right {
  flex: 1;
  width: 50%;
  background-color: #e6f2ff;
  padding-bottom: 10%;
  padding-left: 5%;
}
.box-container {
  display: block;
  width: 70%;
  padding-bottom: 50px;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0px 2px 4px #676565;
}
.logo-header {
  display: flex;
  margin-top: 30px;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  background-color: white;
  margin-bottom: 2px;
}
.inner-box {
  width: 100%;
  padding-left: 20%;
  padding-right: 20%;
  display: flex;
  justify-content: center;
  flex-direction: column;
}

.logo {
  margin-left: 25px;
  width: 79px;
  height: 70px;
}
.product-name {
  line-height: 37px;
  margin-left: 9px;
  font-family: "Poppins", sans-serif;
  font-family: 25px;
  font-weight: bold;
  color: #000000;
}
.common-text {
  line-height: 10px;
  font-family: "Poppins", sans-serif;
  font-size: 15px;
  color: #000000;
  text-align: center;
}
.user-and-pass {
  width: 100%;
  background-color: white;
  text-align: center;
  margin-top: 13px;
}
:placeholder {
  color: #676565;
  font-family: "Roboto", sans-serif;
  font-size: 1px;
}
.textfield-wrap {
  display: flex;
  border-bottom: 1.5px solid #3da5c6;
  padding: 8px 10px;
  margin-bottom: 18px;
  margin-top: 4px;
  margin-left: auto;
  margin-right: auto;
}
.textfield {
  box-sizing: border-box;
  text-align: center;
  border: none;
  outline: none;
}
.rememberMe-wrap {
  display: flex;
  padding: 8px 10px;
  margin-bottom: 12px;
  flex-direction: row;
  align-items: center;
  color: #444444;
  font-weight: bold;
}

.button {
  display: flex;
  margin-top: 44px;
  padding-top: 5px;
  padding-bottom: 5px;
  justify-content: center;
  align-items: center;
  border-radius: 10px;
  background-color: #28b4e0;
  cursor: pointer;
}
.button text {
  color: #ffffff;
  font-size: 20px;
  font-family: poppins;
  font-weight: 500;
}

.font {
  flex: 1;
  padding-left: 61px;
  padding-top: 106px;
  width: 100%;
  padding-bottom: 48px;
}
.font h1 {
  font-family: "Roboto", sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 40px;
  line-height: 46.88px;
  color: #000000;
}
.font h2 {
  font-family: "Poppins", sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 30px;
  line-height: 45px;
  color: #000000;
}
.image {
  flex: 2;
  width: 100%;
}
.image-round {
  width: 416px;
  height: 423px;
  margin-left: 149px;
  resize: inherit;
}

.image-customer {
  width: 427px;
  height: 424px;
  position: absolute;
  top: 25%;
  left: 50%;
}

.switch-signin{
  color: #15aad9;
}