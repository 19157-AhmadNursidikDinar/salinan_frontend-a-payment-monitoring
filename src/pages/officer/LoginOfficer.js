import React, { useState } from "react";
import { withStyles } from "@material-ui/core/styles";
import InputLabel from "@material-ui/core/InputLabel";
import FormControl from "@material-ui/core/FormControl";
import Select from "@material-ui/core/Select";
import TextField from "@material-ui/core/TextField";
import Grid from '@material-ui/core/Grid';
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Checkbox from "@material-ui/core/Checkbox";
import Button from "@material-ui/core/Button";
import Alert from "@material-ui/lab/Alert";
import { Link } from "react-router-dom";

import InputAdornment from "@material-ui/core/InputAdornment";
import PersonIcon from "@material-ui/icons/Person";
import LockIcon from "@material-ui/icons/Lock";

import ColorsTheme from "../../assets/colors";
import Logo from "../../assets/images/logoPM.png";
import BankerLogo from "../../assets/images/Banker.png";
import Circle from "../../assets/images/elipse.png";
import useStyles from "../../styles/LoginPage";

import { useFormik } from "formik";
import * as Yup from "yup";
import AuthService from "../../services/auth.service";

const BlueCheckbox = withStyles({
  root: {
    color: ColorsTheme.blueJeans,
    "&$checked": {
      color: ColorsTheme.bleuDeFrance,
    },
  },
  checked: {},
})((props) => <Checkbox color="default" {...props} />);

const validationSchema = Yup.object().shape({
  username: Yup.string().required("Insert username!"),
  password: Yup.string()
    .min(6, "Use combination of 6 character or more")
    .required("Insert Password"),
  rememberMe: Yup.boolean(),
});

export default function LoginOfficer(props) {
  const classes = useStyles();
  const [errorMsg, setErrorMsg] = useState("");
  const formik = useFormik({
    initialValues: {
      username: "",
      password: "",
      rememberMe: false,
    },
    validationSchema: validationSchema,
    validateOnBlur: false,
    validateOnChange: false,
    onSubmit: async ({ username, password, rememberMe }) => {
      const result = await AuthService.login({
        username,
        password,
        rememberMe,
      });
      // console.log({ result });
      if (!Boolean(result.error)) {
        const role = AuthService.getUserRole();
        if (role === "ADMIN") {
          props.history.push("/admin");
        } else if (role === "GENERAL-SUPPORT") {
          props.history.push("/general-support");
        } else if (role === "ACCOUNTING") {
          props.history.push("/accounting");
        } else if (role === "USER") {
          props.history.push("/customer");
        } else {
          setErrorMsg("undefined role");
        }
      } else {
        setErrorMsg(result.error.response.data.msg);
      }
    },
  });

  return (
    <Grid container className={classes.root}>
      <Grid item xs={12} md={6}>
        <div className={classes.leftSide}>
          <div className={classes.boxContainer}>
            <div className={classes.flexCenter}>
              <div className={classes.headerLogo}>
                <img src={Logo} alt="app-logo" className={classes.headLogo} />
                <h1 className={classes.titleSection}>
                  Pay<span className={classes.titleSectionSpan}>ment</span> <br /> Monitoring
              </h1>
              </div>
            </div>
            <h3 className={classes.txtSignIn}>Sign in to continue our application </h3>
            <div className={classes.innerBox}>
              {errorMsg && (
                <Alert severity="error" style={{ margin: "0.5em 0" }}>
                  {errorMsg}
                </Alert>
              )}
              <form onSubmit={formik.handleSubmit}>
                <div className={classes.userAndPass}>
                  <div className={classes.wrappedTxtFieldOfficer}>
                    <TextField
                      className="txtfield"
                      type="text"
                      placeholder="Username"
                      name="username"
                      value={formik.values.username}
                      onChange={formik.handleChange}
                      variant="outlined"
                      fullWidth
                      InputProps={{
                        startAdornment: (
                          <InputAdornment position="start">
                            <PersonIcon style={{ color: ColorsTheme.cyanProcess }} />
                          </InputAdornment>
                        ),
                      }}
                      disabled={formik.isSubmitting}
                      error={
                        Boolean(formik.errors.username) && formik.touched.username
                      }
                      helperText={formik.errors.username}
                    />
                  </div>
                  <div className={classes.wrappedTxtFieldOfficer}>
                    <TextField
                      name="password"
                      className="txtfield"
                      type="password"
                      placeholder="Password"
                      variant="outlined"
                      fullWidth
                      InputProps={{
                        startAdornment: (
                          <InputAdornment position="start">
                            <LockIcon style={{ color: ColorsTheme.cyanProcess }} />
                          </InputAdornment>
                        ),
                      }}
                      value={formik.values.password}
                      onChange={formik.handleChange}
                      disabled={formik.isSubmitting}
                      error={
                        Boolean(formik.errors.password) && formik.touched.password
                      }
                      helperText={formik.errors.password}
                    />
                  </div>
                  <FormControl variant="outlined" className={classes.formControl}>
                    <InputLabel htmlFor="outlined-age-native-simple">
                      Login As
                  </InputLabel>
                    <Select
                      native
                      // onChange={(e) => {
                      //   setSignInAs(e.target.value);
                      // }}
                      label="Login As"
                      inputProps={{
                        shrink: true,
                        startAdornment: (
                          <InputAdornment position="start">
                            <LockIcon style={{ color: ColorsTheme.cyanProcess }} />
                          </InputAdornment>
                        ),
                      }}
                    // value={signInAs || ""}
                    >
                      <option value="generalSupport">General Support</option>
                      <option value="accounting">Accounting</option>
                      <option value="admin">Admin</option>
                    </Select>
                  </FormControl>
                </div>
                <div className={classes.wrappedRememberMe}>
                  <FormControlLabel
                    control={
                      <BlueCheckbox
                        name="rememberMe"
                        checked={formik.values.rememberMe}
                        onChange={formik.handleChange}
                        disabled={formik.isSubmitting}
                      />
                    }
                    label="Remember Me"
                  />
                </div>
                <div className={classes.wrappedSignIn} fullWidth>
                  <Button
                    className={
                      formik.isSubmitting ? classes.btnSignInLoading : classes.btnSignIn
                    }
                    fullWidth
                    type="submit"
                    disabled={formik.isSubmitting}
                  >
                    <b className={classes.btnSignInBold}>{formik.isSubmitting ? "Loading..." : "Sign In"}</b>
                  </Button>
                </div>
                <div
                  style={{
                    marginTop: 20,
                    justifyContent: "center",
                    display: "flex",
                  }}
                >
                  <Link className={classes.switchSignIn} to="/">
                    Sign In as Customer
                </Link>
                </div>
              </form>
            </div >
          </div >
        </div >
      </Grid>
      <Grid item xs={0} md={6}>
        <div className={classes.rightSide}>
          <div className={classes.welcomeTitle}>
            <h1 className={classes.welcome}>WELCOME BACK,</h1>
            <h2 className={classes.textNotice}>Make your payment efficiently</h2>
          </div>
          <div className={classes.img}>
            <img className={classes.circledImage} src={Circle} alt="Round.png" />
            <img className={classes.bankerImage} src={BankerLogo} alt="BankerLogo" />
          </div>
        </div>
      </Grid>
    </Grid>
  );
}
