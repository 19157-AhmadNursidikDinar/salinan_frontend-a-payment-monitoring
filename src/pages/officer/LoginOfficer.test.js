import React from "react"; 
import { shallow } from "enzyme";
import LoginOfficer from "./LoginOfficer";
import TextField from "@material-ui/core/TextField";

describe("Login Officer". () => {
    const wrapper = shallow(<LoginOfficer/>);
    it("should render Login Officer". () => {
        expect(wrapper).toMatchSnapshot();
    });
    it("should render two TextField". () => {
        expect(wrapper.find(TextField).length).toBe(3);
    });
})