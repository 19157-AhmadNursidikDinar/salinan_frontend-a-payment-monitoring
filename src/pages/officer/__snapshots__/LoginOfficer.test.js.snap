// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Officer should render Login Officer 1`] = `ShallowWrapper {}`;
